src/target/riscv/riscv-013.c

```diff
@@ -606,6 +606,12 @@ static int dmi_op_timeout(struct target *target, uint32_t *data_in,
 					false);
 			if (status == DMI_STATUS_BUSY) {
 				increase_dmi_busy_delay(target);
+				if (dmi_op == DMI_OP_READ) {
```

Comment:
So, there used to be transparent retry at this level when a busy result came back, and now the retry responsibility has been moved upward to the caller(s)?

Reply:
These changes look good to me (especially having read the latest comments).  Sounds like the problem was exposed through stress testing (Spike?), and also the fix verified by the same tests.